presto-tests/src/test/java/com/facebook/presto/server/TestClusterStatsResource.java

```diff
@@ -0,0 +1,98 @@
+import com.facebook.presto.resourceGroups.FileResourceGroupConfigurationManagerFactory;
+import com.facebook.presto.server.testing.TestingPrestoServer;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.facebook.airlift.http.client.JsonResponseHandler.createJsonResponseHandler;
+import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
+import static com.facebook.airlift.json.JsonCodec.jsonCodec;
+import static com.facebook.airlift.testing.Closeables.closeQuietly;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_USER;
+import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunner;
+import static com.facebook.presto.utils.QueryExecutionClientUtil.runToExecuting;
+import static java.lang.Thread.sleep;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestClusterStatsResource
+{
+    private HttpClient client;
+    private TestingPrestoServer server;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        client = new JettyHttpClient();
+        DistributedQueryRunner runner = createQueryRunner(ImmutableMap.of("query.client.timeout", "10s"));
+        server = runner.getCoordinator();
+        server.getResourceGroupManager().get().addConfigurationManagerFactory(new FileResourceGroupConfigurationManagerFactory());
+        server.getResourceGroupManager().get()
+                .setConfigurationManager("file", ImmutableMap.of("resource-groups.config-file", getResourceFilePath("resource_groups_config_simple.json")));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        closeQuietly(server);
+        closeQuietly(client);
+        server = null;
+        client = null;
+    }
+
+    @Test(timeOut = 60_000, enabled = false)
+    public void testGetClusterStats()
+            throws Exception
+    {
+        runToExecuting(client, server, "SELECT * from tpch.sf100.orders");
+
+        // Sleep to allow query to make some progress
+        sleep(SECONDS.toMillis(5));
+
+        Request request = prepareGet()
+                .setHeader(PRESTO_USER, "user")
+                .setUri(uriBuilderFrom(server.getBaseUrl().resolve("/v1/cluster")).build())
+                .setFollowRedirects(true)
+                .build();
+
+        ClusterStatsResource.ClusterStats clusterStats = client.execute(request, createJsonResponseHandler(jsonCodec(ClusterStatsResource.ClusterStats.class)));
+
+        assertNotNull(clusterStats);
+        assertTrue(clusterStats.getActiveWorkers() > 0);
```

Comment:
it should be a fixed value, i think in this case it will be 4 workers + 1 coordinator so a total of 5.

Reply:
Updated the test